packages/langium/test/package-main.tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "../tsconfig.src.json",
+    "compilerOptions": {
+      "noEmit": true
+    },
+    "exclude": [
+        "../src/test/**/*"
+    ]
+}
```

Comment:
That's very smart 🎩

I'm just confused by the name and location of the file. What about
`tsconfig.src-main.json`,  `tsconfig.src-main-only.json`, `tsconfig.src-plain.json`, `tsconfig.src-plain-only.json`? (to have it inline with the other `tsconfig`s).

I would love to see a comment like
`// only used validating to not import modules from 'src/test', ... within base modules like 'src/parser/*'
in line 1 😉

Reply:
> I'm just confused by the name and location of the file. What about
tsconfig.src-main.json, tsconfig.src-main-only.json, tsconfig.src-plain.json, tsconfig.src-plain-only.json? (to have it in line with the other tsconfigs).

I'd prefer to keep it separate to highlight that this is part of the framework tests and that it's not used for building. Having too many tsconfigs next to each other could be confusing.

> I would love to see a comment like `// only used for validating to not import modules from 'src/test', ... within base modules like 'src/parser/*' in line 1 😉

I got an error in the JSON editor when I added a comment (comments are not part of the standard JSON syntax).

Reply:
That's another pro arg for renaming, as `tsconfig. ... .json` files are handling differently. With such a name, comments are permitted, see
<img width="871" alt="image" src="https://github.com/eclipse-langium/langium/assets/8513883/cbce83fc-abce-4753-ab15-9ccef874f1df">

Reply:
Ok, renamed and added a comment. I still think it's better to keep the file in a subfolder to avoid confusion.

Reply:
🎉